src/components/TopView.tsx

```diff
@@ -1,95 +1,315 @@
-  };
-  const styles = useStyles();
-
-  return (
-    <div className={styles.topView}>
-      <Paper className={styles.topView + ' ' + styles.paper}>
-        <TabContext value={String(value)}>
-          <Tabs textColor="primary" value={value} onChange={handleChange}>
-            <Tab label="Data Visualization" />
-            <Tab label="File Upload" />
-            <Tab label="Tutorials" />
-          </Tabs>
-          <TabPanels className={styles.tabPanel} value={value}>
-            <DataView />
-            <Paper variant="outlined">
-              <FileUpload />
+    const [value, setValue] = useState(0);
+    const [datasets, setDatasets] = useState([]);
+
+    const { enqueueSnackbar } = useSnackbar();
+
+    const handleChange = (event: any, newValue: number) => {
+        setValue(newValue);
+    };
+
+    const [user] = useAuthState(firebase.auth());
+
+    const styles = useStyles();
+
+    useEffect(() => {
+        async function getDataSets() {
+            fetch(process.env.NEXT_PUBLIC_GQL_API_UPDATED, {
+                method: "POST",
+                headers: {
+                    "Content-Type": "application/json",
+                    Authorization: (await getToken()) as unknown as string
+                },
+                body: JSON.stringify({
+                    query: `query GetUser($id: String!) {
+                           user(id: $id) {
+                                 id
+                                 bio
+                                 url
+                                 datasets
+                            }
+                        }
+                    `,
+                    variables: {
+                        id: user.uid
+                    }
+                })
+            })
+                .then((resp) => resp.json())
+                .then((result) => setDatasets(result.data.user.datasets));
+        }
+        globalThis.datasets = datasets;
+        getDataSets();
+    }, [datasets, user.uid]);
+
+    async function getToken() {
+        const token = await firebase.auth().currentUser?.getIdToken();
+
+        if (token === undefined) return {};
+
+        return token;
+    }
+
+    return (
+        <div className={styles.topView}>
+            <Paper
+                style={{ width: "100%!important" }}
+                className={styles.topView + " " + styles.paper}
+            >
+                <TabContext value={String(value)}>
+                    <Tabs
+                        variant="fullWidth"
+                        textColor="primary"
+                        value={value}
+                        onChange={handleChange}
+                    >
+                        <Tab label="Data Visualization" />
+                        <Tab label="Data sets" />
+                    </Tabs>
+                    <TabPanels className={styles.tabPanel} value={value}>
+                        <DataView />
+                        <div style={{ overflow: "scoll" }}>
```

Comment:
You should move this whole thing into a new component, and also for doing CSS it is a good idea to use material ui's makestyles system (let me know if you want me to show you how it works).